datalake/src/main/java/com/sequenceiq/datalake/flow/delete/handler/StackDeletionHandler.java

```diff
@@ -72,6 +89,27 @@ protected Selectable doAccept(HandlerEvent<StackDeletionWaitRequest> event) {
             LOGGER.error("Deletion polling failed for stack: {}", sdxId);
             response = new SdxDeletionFailedEvent(sdxId, userId, exception, stackDeletionWaitRequest.isForced());
         }
+        deleteStructuredEvents(response, resourceCrn, sdxId);
         return response;
     }
+
+    private void deleteStructuredEvents(Selectable terminationOutcome, Optional<String> resourceCrn, Long sdxId) {
```

Comment:
My only concern with this, from a resize perspective, is that this seems like it will delete the structured events for the original datalake.

Specifically, resize does roughly the following: Detach original DL (mainly metadata changes), Create new DL, Delete original DL. So would this not delete the structured events associated with the original DL? This seems not ideal since customers might lose old events that ideally should remain since the DL effectively still exists.

Reply:
Maybe we could have an additional check for performing this operation which checks if the deletion is part of the resize process? For resize this would be as simple as checking if the DL being deleted has its internal `detached` flag set to `true`.

We may also need something similar for resize recovery, where the new DL is deleted because either its creation or data recovery failed. In this case the `detached` flag will not be usable since we do not update it. However in both cases we do set the `originalCrn` field in the DL's to something, so this field could be used instead.